hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSource.java

```diff
@@ -224,6 +226,35 @@ public void init(Configuration conf, FileSystem fs, Path walDir,
     this.abortOnError = this.conf.getBoolean("replication.source.regionserver.abort",
       true);
 
+    if (conf.getBoolean(HConstants.REPLICATION_OFFLOAD_ENABLE_KEY,
+      HConstants.REPLICATION_OFFLOAD_ENABLE_DEFAULT)) {
+      fetchWALsThread = new Thread(() -> {
```

Comment:
interrrupt the `fetchWALsThread` when the ReplicationServer exits or when the peer terminates?